electrumsv/tests/test_wallet.py

```diff
@@ -608,6 +613,142 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
+            if sys.version_info[:3] < (3, 8, 0):
+                if "The specified module could not be found" in e.args[1]:
+                    pytest.xfail("libusb DLL could not be found")
+                    return
+            raise e
+
+    wallet = load_wallet(storage)
+    if not wallet:
+        return
+
+    # Store pre-password update related data to compare against post-password update data.
+    private_keys_before: dict[str, str] = {}
+    assert len(wallet.get_accounts()) == 2
+    private_key_account = cast(StandardAccount,
+        [entry for entry in wallet.get_accounts() if not entry.is_petty_cash()][0])
+    private_key_keystore = cast(BIP32_KeyStore, private_key_account.get_keystore())
+    xprv = private_key_keystore.get_master_private_key(initial_password)
+    privkey = bitcoinx.bip32_key_from_string(xprv)
+    public_key = privkey.public_key
+    private_keys_before[public_key.to_hex()] = privkey.to_hex()
+
+    encrypted_api_keys_before: list[str] = []
+    for server in wallet.data.read_network_servers():
+        if server.encrypted_api_key is not None:
+            encrypted_api_keys_before.append(pw_decode(server.encrypted_api_key, initial_password))
+
+    password = "654321"
+    future, update_completion_event = wallet.update_password(initial_password, password)
+    # Wait for the database update to finish.
+    future.result(5)
+    # Wait for the done callback to finish.
+    update_completion_event.wait()
+
+    # Perform a full shutdown and reload of the wallet database. This will cover the code pathways
+    # of:
+    # - flushing the network server cache updates to the database (which was causing a bug where
+    # it was overwriting the updated encrypted api key with the old version).
+    # - reloading of the `WalletData` table variables into the wallet cache (which involves json
+    # deserialization of stored values)
+    # - loading the `Server` table data and decrypting the api keys with the new password
+    wallet.stop()
+    storage = WalletStorage(wallet_path)
+    wallet = load_wallet(storage)
+    if not wallet:
+        return
+
+    # Check `master_private_key` data decryption with new password
+    private_keys_after: dict[str, str] = {}
+    xprv = private_key_keystore.get_master_private_key(password)
+    privkey = bitcoinx.bip32_key_from_string(xprv)
+    public_key = privkey.public_key
+    private_keys_after[public_key.to_hex()] = privkey.to_hex()
+
+    # Check `encrypted_api_key` data decryption with new password
+    encrypted_api_keys_after: list[str] = []
+    for server in wallet.data.read_network_servers():
+        if server.encrypted_api_key is not None:
+            encrypted_api_keys_after.append(pw_decode(server.encrypted_api_key, password))
+
+    assert private_keys_before == private_keys_after
+    assert encrypted_api_keys_before == encrypted_api_keys_after
+
+    check_specific_wallets(wallet, password, storage_info)
+
+    if expected_network in {"regtest"}:
```

Comment:
So what happens if the test fails?

Reply:
Okay, I can wrap the test in a try/finally block to ensure the `Net.set_to(SVMainnet)` always executes. But I note that this is not done for `test_legacy_wallet_loading` - does the same change need to be made there too?

Reply:
Actually, I will do it with a fixture like this to avoid indenting all of the code...

```
@pytest.fixture()
def set_and_unset_regtest_network(request):
    Net.set_to(SVRegTestnet)
    try:
        yield
    finally:
        Net.set_to(SVMainnet)
```